src/triage/component/postmodeling/error_analysis.py

```diff
@@ -0,0 +1,169 @@
+   error_labeler (Y).
+   Arguments:
+       - depth: max number of tree partitions. This is passed directly to
+         the classifier.
+       - view_plot: the plot is saved to disk by default, but the
+         graphviz.Source also allow to load the object and see it in the
+         default OS image renderer
+       - **kwargs: more arguments passed to the labeler: param indicating
+         the threshold value, param_type indicating the type of threshold,
+         and the path to the ProjectStorage.
+   '''
+
+   # Get matrices from the labeler
+   zip_data = self._error_labeler(param_type = kwargs['param_type'],
+                              param = kwargs['param'],
+                              path=kwargs['path'])
+
+   # Model tree and output tree plot
+   for error_label, matrix in zip_data:
+
+       dot_path = 'error_analysis_' + \
+                  str(error_label[1]) + '_' + \
+                  str(self.model_id) + '_' + \
+                  str(kwargs['param_type']) + '@'+ \
+                  str(kwargs['param']) +  '.gv'
+
+       clf = tree.DecisionTreeClassifier(max_depth=depth)
+       clf_fit = clf.fit(matrix, error_label[0])
+       tree_viz = tree.export_graphviz(clf_fit,
+                                       out_file=None,
+                                       feature_names=matrix.columns.values,
+                                       filled=True,
+                                       rounded=True,
+                                       special_characters=True)
+       graph = graphviz.Source(tree_viz)
+       graph.render(filename=dot_path,
+                    directory='error_analysis',
+                    view=view_plots)
+
+       print(dot_path)
+
+def error_analysis(self, threshold, **kwargs):
+    '''
+    Error analysis function for ThresholdIterator objects. This function
+    have the same functionality as the _error.modeler method, but its
+    implemented for iterators, which can be a case use of this analysis.
+    If no iterator object is passed, the function will take the needed
+    arguments to run the _error_modeler.
+    Arguments:
+        - threshold: a threshold and threshold parameter combination passed
+        to the PostmodelingParamters. If multiple parameters are passed,
+        the function will iterate through them.
+        -**kwags: other arguments passed to _error_modeler
+    '''
+
+    error_modeler = partial(self._error_modeler,
+                           depth = kwargs['depth'],
+                           path = kwargs['path'],
+                           view_plots = kwargs['view_plots'])
+
+    if isinstance(threshold, dict):
+       for threshold_type, threshold_list in threshold.items():
+           for threshold in threshold_list:
+               print(threshold_type, threshold)
+               error_modeler(param_type = threshold_type,
+                             param = threshold)
+    else:
```

Comment:
I'm confused about the `threshold` param -- it's required, but if it's not a `dict`, it's completely ignored?